apitools/org.eclipse.pde.api.tools/META-INF/MANIFEST.MF

```diff
@@ -35,7 +33,10 @@ Export-Package: org.eclipse.pde.api.tools.internal;x-friends:="org.eclipse.pde.a
  org.eclipse.pde.api.tools.internal.util;x-friends:="org.eclipse.pde.api.tools.tests,org.eclipse.pde.api.tools.ui,org.eclipse.pde.api.tools.generator",
  org.eclipse.pde.api.tools.internal.util.profiles;x-internal:=true
 Bundle-RequiredExecutionEnvironment: JavaSE-11
-Import-Package: com.ibm.icu.util
+Import-Package: com.ibm.icu.util,
+ org.objectweb.asm;version="[9.2.0,9.4.0)",
```

Comment:
Make these all `[9.3.0,9.4.0)`?

Reply:
Don't know, it previously has required something from 9.2 and 9.3 so maybe it was meant to be compatible with both... so if upgrading from 9.2 > 9.3 does not required any code changes the lower bound would be fine here.

Reply:
9.3 is required for Java 19 work.

Reply:
The intersection of the require bundle versions is `[9.3.0,10.0.0)` and this is what you want for the import package ranges of the asm packages (since ASM seems to version their bundles and packages identically). `[9.2.0,9.4.0)` or `[9.3.0,9.4.0)` is excessively narrow.

Reply:
I think the 9.3 lower bound update on asm.tree was simply missed in https://github.com/eclipse-pde/eclipse.pde/commit/404eee00f455b45956705070906f40726c2219a2

Reply:
For example `ReferenceExtractor` extends `ClassVisitor` what is an ASM class (there are more) so we need to use provider version range I think here what would be `[9.3.0,9.4.0)`?

Reply:
> For example `ReferenceExtractor` extends `ClassVisitor` what is an ASM class (there are more) so we need to use provider version range I think here what would be `[9.3.0,9.4.0)`?

In that case you are right. Even if it requires more version bumps, but it has to be done.

Reply:
> For example `ReferenceExtractor` extends `ClassVisitor` what is an ASM class (there are more) so we need to use provider version range I think here what would be `[9.3.0,9.4.0)`?

But no one cared about this before since the bundle-version was [9.3.0,10.0.0) :-)

You are not a provider of the ASM API. You are a consumer of the API and you can rely upon the provider of the API (ASM) to not break consumer types like ClassVisitor without a major version bump. See <https://docs.osgi.org/whitepaper/semantic-versioning/060-importer-policy.html>.